pkg/sync/syncer.go

```diff
@@ -1828,6 +1828,35 @@ func (s *syncer) processGrantsWithExternalPrincipals(ctx context.Context, princi
 	expandedGrants := make([]*v2.Grant, 0)
 	for _, grant := range grants {
 		annos := annotations.Annotations(grant.Annotations)
+		matchResourceMatchIDAnno, err := GetExternalResourceMatchIDAnnotation(annos)
+		if err != nil {
+			return err
+		}
+		if matchResourceMatchIDAnno != nil {
+			for _, principal := range principals {
```

Comment:
How many grants & principals do we expect?

Reply:
I'm not sure, I'll use a map